addons/persistence/jdbc-persistence-addon/src/main/java/org/kie/kogito/persistence/jdbc/JDBCProcessInstances.java

```diff
@@ -0,0 +1,265 @@
+
+    public JDBCProcessInstances(Process<?> process, DataSource dataSource, boolean autoDDL,
+            String proto,
+            BaseMarshaller<?>... marshallers) {
+        this.dataSource = dataSource;
+        this.process = process;
+        this.autoDDL = autoDDL;
+        marshaller = new ProcessInstanceMarshaller(new ProtoStreamObjectMarshallingStrategy(proto, marshallers));
+        init();
+    }
+
+    private void init() {
+        if (!autoDDL) {
+            LOGGER.debug("Auto DDL is disabled, do not running initializer scripts");
+            return;
+        }
+
+        try (Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getQueryFromFile("exists_tables"))) {
+            createTable(connection, statement);
+
+        } catch (Exception e) {
+            //not break the execution flow in case of any missing permission for db application user, for instance.
+            LOGGER.error("Error creating process_instances table, the database should be configured properly before " +
+                    "starting the application", e);
+        }
+    }
+
+    private void createTable(Connection connection, PreparedStatement statement) {
+        boolean result = false;
+        try (ResultSet resultSet = statement.executeQuery()) {
+            if (resultSet.next()) {
+                result = Optional.ofNullable(resultSet.getBoolean("exists"))
+                        .filter(Boolean.FALSE::equals)
+                        .map(e -> {
+                            try {
+                                PreparedStatement prepareStatement = connection.prepareStatement(getQueryFromFile("create_tables"));
+                                return prepareStatement.execute();
+                            } catch (SQLException e1) {
+                                LOGGER.error("Error creating process_instances table", e1);
+                            }
+                            return false;
+                        })
+                        .orElseGet(() -> {
+                            LOGGER.info("Table process_instances already exists.");
+                            return false;
+                        });
+            }
+
+        } catch (Exception e) {
+            throw uncheckedException(e, "Error creating process_instances table");
+        }
+
+        if (result) {
+            LOGGER.info("DDL successfully done for ProcessInstance");
+        } else {
+            LOGGER.info("DDL executed with no changes for ProcessInstance");
+        }
+    }
+
+    private String getQueryFromFile(String scriptName) {
+        try {
+            return new String(IoUtils.readBytesFromInputStream(
+                    Thread.currentThread()
+                            .getContextClassLoader()
+                            .getResourceAsStream(String.format("sql/%s.sql", scriptName))));
```

Comment:
Suggesting the following to avoid dependency on v7 API
```suggestion
            String path = Thread.currentThread().getContextClassLoader().getResource(String.format("sql/%s.sql", scriptName)).getPath();
            return new String(Files.readAllBytes(Paths.get(path)));
```

Reply:
@tiagodolphine sent a PR for that and some other usage here https://github.com/kiegroup/kogito-runtimes/pull/1295

Reply:
done.